src/Mod/PartDesign/Gui/TaskChamferParameters.cpp

```diff
@@ -328,6 +346,96 @@ void TaskChamferParameters::apply()
+    }
+
+    gizmos = std::make_unique<Gizmos>();
+
+    auto distanceGizmo = new Gui::LinearGizmo(ui->chamferSize);
+    auto twoDistanceGizmo = new Gui::LinearGizmo(ui->chamferSize);
+    auto angleGizmo = new Gui::RotationGizmo(ui->chamferAngle);
+
+    connect(ui->chamferType, qOverload<int>(&QComboBox::currentIndexChanged), [this, twoDistanceGizmo, angleGizmo] (int index) {
+        switch (index) {
+            case 0:
+                twoDistanceGizmo->getDraggerContainer()->visible = true;
+                angleGizmo->getDraggerContainer()->visible = false;
+
+                twoDistanceGizmo->setProperty(ui->chamferSize);
+                twoDistanceGizmo->setDragLength(ui->chamferSize->value().getValue());
+
+                break;
+            case 1:
+                twoDistanceGizmo->getDraggerContainer()->visible = true;
+                angleGizmo->getDraggerContainer()->visible = false;
+
+                twoDistanceGizmo->setProperty(ui->chamferSize2);
+                twoDistanceGizmo->setDragLength(ui->chamferSize2->value().getValue());
+
+                break;
+            case 2:
+                twoDistanceGizmo->getDraggerContainer()->visible = false;
+                angleGizmo->getDraggerContainer()->visible = true;
+
+        }
+    });
+
+    connect(ui->flipDirection, &QCheckBox::toggled, [distanceGizmo, twoDistanceGizmo] (bool) {
+        GizmoPlacement placement = distanceGizmo->getDraggerPlacement();
+        GizmoPlacement placement2 = twoDistanceGizmo->getDraggerPlacement();
+
+        distanceGizmo->setDraggerPlacement(placement2.pos, placement2.dir);
+        twoDistanceGizmo->setDraggerPlacement(placement.pos, placement.dir);
+    });
+
+    gizmos->addGizmo(distanceGizmo);
+    gizmos->addGizmo(twoDistanceGizmo);
+    gizmos->addGizmo(angleGizmo);
+    gizmos->initGizmos();
+
+    setGizmoPositions();
+
+    vp->attachGizmos(gizmos.get());
+}
+
+void TaskChamferParameters::setGizmoPositions()
+{
+    if (!gizmos) {
+        return;
+    }
+
+    auto distanceGizmo = gizmos->getGizmo<LinearGizmo>(0);
+    auto twoDistanceGizmo = gizmos->getGizmo<LinearGizmo>(1);
+    auto angleGizmo = gizmos->getGizmo<RotationGizmo>(2);
+
+    auto chamfer = getObject<PartDesign::Chamfer>();
+    auto baseShape = chamfer->getBaseTopoShape();
+    auto shapes = chamfer->getContinuousEdges(baseShape);
+
+    if (shapes.size() == 0) {
+        gizmos->visible = false;
+        return;
+    } else {
+        gizmos->visible = true;
+    }
+
+    Part::TopoShape edge = shapes[0];
+    auto [face1, face2] = getAdjacentFacesFromEdge(edge, baseShape);
+
+    DraggerPlacementProps props = getDraggerPlacementFromEdgeAndFace(edge, face1);
+    distanceGizmo->Gizmo::setDraggerPlacement(props.position, props.dir);
+
+    angleGizmo->placeBelowLinearGizmo(distanceGizmo);
+    angleGizmo->getDraggerContainer()->setArcNormalDirection(Base::convertTo<SbVec3f>(-props.tangent));
+    angleGizmo->getDraggerContainer()->visible = getType() == 2;

```

Comment:
`getType() == 2` does not have meaning while reading, it should use enum or at least have a comment if it does not have anum.

Reply:
Added a comment